Make sure we have enough @types non-dev dependencies

Any packages whose types end up in our public API (say, in the .d.ts files) need to have non-dev `@types/` dependencies.  See https://github.com/apollographql/apollo-engine-js/issues/159 for an example of messing this up in another app.

This is pretty easy to reproduce: just try to use published packages inside an empty TS project.  eg:

```
glasser@glasser-mbp 2 /tmp/zz $ cat package.json
{
  "name": "zz",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "apollo-link": "^1.2.1",
    "typescript": "^2.7.2"
  }
}
glasser@glasser-mbp 0 /tmp/zz $ cat x.ts
import {from} from 'apollo-link';

from([]);

glasser@glasser-mbp 0 /tmp/zz $ npx tsc x.ts
node_modules/apollo-link/lib/types.d.ts(2,47): error TS2307: Cannot find module 'graphql'.
```

--- a/packages/apollo-link/CHANGELOG.md
+++ b/packages/apollo-link/CHANGELOG.md
@@ -1,9 +1,10 @@
 # Change log
 
 ### vNEXT
+- export graphql types and add @types/graphql as a regular dependency [PR#576](https://github.com/apollographql/apollo-link/pull/576)
 
 ### 1.2.1
-- udate apollo link with zen-observable-ts to remove import issues [PR#515](https://github.com/apollographql/apollo-link/pull/515)
+- update apollo link with zen-observable-ts to remove import issues [PR#515](https://github.com/apollographql/apollo-link/pull/515)
 
 ### 1.2.0
 - Add `fromError` Observable helper

--- a/packages/apollo-link/package.json
+++ b/packages/apollo-link/package.json
@@ -23,14 +23,17 @@
   },
   "homepage": "https://github.com/apollographql/apollo-link#readme",
   "scripts": {
-    "build:browser": "browserify ./lib/bundle.umd.js -o=./lib/bundle.js --i apollo-utilities --i graphql --i zen-observable-ts && npm run minify:browser",
+    "build:browser":
+      "browserify ./lib/bundle.umd.js -o=./lib/bundle.js --i apollo-utilities --i graphql --i zen-observable-ts && npm run minify:browser",
     "build": "tsc -p .",
     "bundle": "rollup -c",
     "clean": "rimraf lib/* && rimraf coverage/*",
     "coverage": "jest --coverage",
     "filesize": "npm run build && npm run build:browser",
-    "lint": "tslint --type-check -p tsconfig.json -c ../../tslint.json src/*.ts",
-    "minify:browser": "uglifyjs -c -m -o ./lib/bundle.min.js -- ./lib/bundle.js",
+    "lint":
+      "tslint --type-check -p tsconfig.json -c ../../tslint.json src/*.ts",
+    "minify:browser":
+      "uglifyjs -c -m -o ./lib/bundle.min.js -- ./lib/bundle.js",
     "postbuild": "npm run bundle",
     "prebuild": "npm run clean",
     "prepublishOnly": "npm run clean && npm run build",
@@ -38,6 +41,7 @@
     "watch": "tsc -w -p . & rollup -c -w"
   },
   "dependencies": {
+    "@types/graphql": "0.12.6",
     "@types/node": "^9.4.6",
     "apollo-utilities": "^1.0.0",
     "zen-observable-ts": "^0.8.6"
@@ -46,7 +50,6 @@
     "graphql": "^0.11.3 || ^0.12.3 || ^0.13.0"
   },
   "devDependencies": {
-    "@types/graphql": "0.12.6",
     "@types/jest": "21.1.10",
     "browserify": "16.1.1",
     "graphql": "0.13.2",
@@ -64,12 +67,7 @@
       ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
     },
     "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
-    "moduleFileExtensions": [
-      "ts",
-      "tsx",
-      "js",
-      "json"
-    ],
+    "moduleFileExtensions": ["ts", "tsx", "js", "json"],
     "mapCoverage": true
   }
 }

--- a/packages/apollo-link/src/types.ts
+++ b/packages/apollo-link/src/types.ts
@@ -1,5 +1,6 @@
 import Observable from 'zen-observable-ts';
 import { ExecutionResult, DocumentNode } from 'graphql';
+export { ExecutionResult, DocumentNode };
 
 export interface GraphQLRequest {
   query: DocumentNode;

